build/release.mk

```diff
@@ -3,15 +3,30 @@ dist: | check-style test package
 
 build-linux:
 	@echo Build Linux amd64
-	env GOOS=linux GOARCH=amd64 $(GO) install $(GOFLAGS) -trimpath -ldflags '$(LDFLAGS)' ./...
+ifeq ($(BUILDER_GOOS_GOARCH),"linux_amd64")
+	env GOOS=linux GOARCH=amd64 $(GO) build -o $(GOBIN) $(GOFLAGS) -trimpath -ldflags '$(LDFLAGS)' ./...
```

Comment:
Not strictly related to this PR, but what is the advantage of using this directory structure? Why just always use `$OSNAME/$BINARYNAME`?

Reply:
Not really sure. But this was already there, I just added this code so that it conforms to the structure expected in `make package`. I guess `git blame` should tell us who to ask this question.

Reply:
I was added in 7bc8e9a08dfde56387f946fdf5086252aa4d0491

@crspeller Would you be fine use `$OSNAME/$BINARYNAME` as the output schema for all platform?

Reply:
The reason `go install` does this by default is so the binaries corresponding to your platform will be accessible in your `PATH` but the cross compiled ones will not. Useful for developing locally.

Reply:
I see. Given that with this PR the output directory is not longer in `$PATH` are you fine with my proposed approach?

Reply:
While I am fine with the changes proposed, I think it would be going slightly out of scope for this PR. I would prefer to keep the changes minimal to let it soak in the CI and smoothen any further wrinkles before implementing this.